Detectors/EMCAL/calibration/run/runCalibOffline.cxx

```diff
@@ -0,0 +1,160 @@
+    }
+
+    if (vm.count("CalibInputPath")) {
+      std::cout << "CalibInputPath was set to "
+                << vm["CalibInputPath"].as<std::string>() << ".\n";
+      CalibInputPath = vm["CalibInputPath"].as<std::string>();
+    } else {
+      std::cout << "CalibInputPath was not set...\n";
+    }
+
+    if (vm.count("ccdbServerPath")) {
+      std::cout << "ccdbServerPath was set to "
+                << vm["ccdbServerPath"].as<std::string>() << ".\n";
+      ccdbServerPath = vm["ccdbServerPath"].as<std::string>();
+    } else {
+      printf("ccdbServerPath was not set.\nWill use standard path %s", ccdbServerPath.c_str());
+    }
+
+    if (vm.count("mode")) {
+      std::cout << "mode was set to "
+                << vm["mode"].as<std::string>() << ".\n";
+      std::string smode = vm["mode"].as<std::string>();
+      if (smode.find("time") != std::string::npos) {
+        std::cout << "performing time calibration" << std::endl;
+        doBadChannelCalib = false;
+      } else if (smode.find("badchannel") != std::string::npos) {
+        std::cout << "performing bad channel calibration" << std::endl;
+        doBadChannelCalib = true;
+      } else {
+        std::cout << "mode not set... returning\n";
+        return 0;
+      }
+    }
+
+    if (vm.count("nameInputHisto")) {
+      std::cout << "nameInputHisto was set to "
+                << vm["nameInputHisto"].as<std::string>() << ".\n";
+      nameCalibInputHist = vm["nameInputHisto"].as<std::string>();
+    }
+
+    if (vm.count("timestampStart")) {
+      std::cout << "timestampStart was set to "
+                << vm["timestampStart"].as<unsigned long>() << ".\n";
+      rangestart = vm["timestampStart"].as<unsigned long>();
+    }
+
+    if (vm.count("timestampEnd")) {
+      std::cout << "timestampEnd was set to "
+                << vm["timestampEnd"].as<unsigned long>() << ".\n";
+      rangeend = vm["timestampEnd"].as<unsigned long>();
+    }
+
+  } catch (bpo::error& e) {
+    std::cerr << "ERROR: " << e.what() << std::endl
+              << std::endl;
+    std::cerr << opt_general << std::endl;
+    exit(1);
+  } catch (std::exception& e) {
+    std::cerr << e.what() << ", application will now exit" << std::endl;
+    exit(2);
+  }
+
+  // Set input file and get histogram
+  TFile* fTimeCalibInput = TFile::Open(CalibInputPath.c_str());
+  if (!fTimeCalibInput) {
+    printf("%s not there... returning\n", CalibInputPath.c_str());
+    return 0;
+  }
+
+  TH2D* hCalibInputHist_ROOT = (TH2D*)fTimeCalibInput->Get(nameCalibInputHist.c_str());
+  if (!hCalibInputHist_ROOT) {
+    printf("%s not there... returning\n", nameCalibInputHist.c_str());
+    return 0;
+  }
+
+  // instance of the calib extractor
+  o2::emcal::EMCALCalibExtractor CalibExtractor;
+
+  // get boost histo from root input histogram
+  auto hCalibInputHist = o2::utils::boostHistoFromRoot_2D(hCalibInputHist_ROOT);
+
+  // instance of CalibDB
+  o2::emcal::CalibDB calibdb(ccdbServerPath);
```

Comment:
@sawenzel @shahor02 Currently the o2::emcal::CalibDB is still using o2::ccdb::CcdbApi as underlying CCDB API. In principle it would be nice to migrate to o2::ccdb::BasicCcdbManager for the CCDB access, but the BasicCcdbManager does not provide write functions, which is i.e. used in this case. Do you see a problem adding a function storing objects which forwards to StoreAsTFileAny to the BasicCcdbManager? The goal of the o2::emcal::CalibDB is to provide a specific interface for EMCAL CDB access so users don't need to mess around with paths.

Reply:
@mfasDa the custom access to the CCDB from the user code in the workflows should be anyway outphased once the DPL CCDB service becomes functional.